mlir/lib/IR/AsmPrinter.cpp

```diff
@@ -1549,6 +1552,18 @@ void SSANameState::numberValuesInRegion(Region &region) {
     if (Operation *op = region.getParentOp()) {
       if (auto asmInterface = dyn_cast<OpAsmOpInterface>(op))
         asmInterface.getAsmBlockArgumentNames(region, setBlockArgNameFn);
+      if (!opAsmOpInterfaceUsed) {
+        // If the OpAsmOpInterface didn't set a name, get name from the type.
+        for (auto arg : region.getArguments()) {
+          if (auto typeInterface =
+                  mlir::dyn_cast<OpAsmTypeInterface>(arg.getType())) {
+            auto setNameFn = [&](StringRef name) {
+              setBlockArgNameFn(arg, name);
+            };
+            typeInterface.getAsmName(setNameFn);
+          }
+        }
```

Comment:
```suggestion
        for (BlockArgument arg : region.getArguments()) {
          if (auto interface = dyn_cast<OpAsmTypeInterface>(arg.getType())) {
            typeInterface.getAsmName([&](StringRef name) {
              setBlockArgNameFn(arg, name);
            });
          }
        }
```

Some style comments related to LLVM style.